src/tbb/scheduler_common.h

```diff
@@ -339,10 +339,30 @@ class tbb_exception_ptr {
     //! Throws the contained exception .
     void throw_self();
 
+    //! Throws the contained exception and then destroys this object.
+    void throw_self_and_destroy();
+
 private:
     tbb_exception_ptr(const std::exception_ptr& src) : my_ptr(src) {}
 }; // class tbb_exception_ptr
 
+//! Utility function to atomically clear and handle exceptions from task_group_context
+/** This function implements thread-safe pattern for clearing exceptions
+    from a task_group_context and either destroying or throwing them. **/
+inline bool handle_context_exception(d1::task_group_context& ctx) {
+    tbb_exception_ptr* exception = ctx.my_exception.load(std::memory_order_acquire);
+    while (exception != nullptr) {
+        // Atomically clear the exception if it's still the same one
+        if (ctx.my_exception.compare_exchange_weak(exception, nullptr, 
+                                                   std::memory_order_acq_rel,
+                                                   std::memory_order_acquire)) {
+            exception->throw_self_and_destroy();
+            return true;
+        }
+    }
```

Comment:
The code contradicts the comment, Since `exception` is updated by `compare_exchange_weak` if CAS fails, the loop actually continues until CAS succeeds - even if the exception is not "the same one" that was there initially.

Is it possible for a new exception to be recorded to the context while this function is in progress? If not, and so the function can only be concurrent to itself, then the loop is unnecessary as `my_exception` can only become `nullptr`. If it is possible, and a new exception has been added concurrently, should it be "handled" by the call?